standard-cluster-services/src/main/java/org/terracotta/entity/IEntityMessenger.java

```diff
@@ -39,6 +39,18 @@
   void messageSelf(EntityMessage message) throws MessageCodecException;
 
   /**
+   * Defer a message based on another message, which will be sent in the future.
```

Comment:
Generally, the requirement from TCStore/Ehcache was to be able to defer retirement to some point in the future, under entity control. We chose to implement this as a decomposition of the existing deferment, which was "here are these 2 messages, one depends on this other one, and send the other one now". This basically gives you a way to setup the deferment dependency without having to send the later message yet. 

Shorter answer, this was a feature we were explicitly asked for.